packages/abi-typegen/src/templates/utils/formatEnums.ts

```diff
@@ -25,7 +25,8 @@ export function formatEnums(params: { types: IType[] }) {
         outputNativeValues,
         typeAnnotations,
       };
-    });
+    })
+    .sort((a, b) => (a.structName < b.structName ? -1 : 1));
```

Comment:
I'm sorting enums and structs to get the same resulting `.d.ts` for when the `--json-abi-with-callpaths` flag is active and not. This is so that I can reuse the [test](https://github.com/FuelLabs/fuels-ts/pull/2372/files#diff-7e78de98857b1c1d9877c6eec968b44cc4e968970a7eccbe1e64cc04ad381e53R17).